odxtools/cli/decode.py

```diff
@@ -30,7 +30,8 @@ def print_summary(
     decode: bool = False,
 ) -> None:
     ecu_names = ecu_variants if ecu_variants else [ecu.short_name for ecu in odxdb.ecus]
-    services: Dict[DiagService, List[str]] = {}
+    service_db: Dict[str, DiagService] = {}
```

Comment:
why not use a named list?

Reply:
for the `service_db` this would indeed be possible, albeit it would change the entry creation code and I cannot see how item access could benefit either. Also, the ` service_ecus` variable cannot easily be changed, because there is no easy way of associating a name to an entry (which is a list of ecu names that feature that service).

The reason why this was necessary in the first place is that I figured that it is not a terribly good idea for `DiagService` objects to be hashable...